lib/fluent/plugin/formatter_csv.rb

```diff
@@ -52,11 +57,11 @@ def configure(conf)
         @generate_opts = {col_sep: @delimiter, force_quotes: @force_quotes, headers: @fields,
                           row_sep: @add_newline ? :auto : "".force_encoding(Encoding::ASCII_8BIT)}
         # Cache CSV object per thread to avoid internal state sharing
-        @cache = {}
+        Thread.current[thread_key] = nil
```

Comment:
Just a note.

>         # Cache CSV object per thread to avoid internal state sharing
>        Thread.current[thread_key] = nil

Now, this code would be unnecessary since the key becomes unique for the instance.
However, it may be the correct specification to clear the cache when `configure` is performed.
(I don't know if there is a case for performing `configure` multiple times.)